lightningd/lightningd.c

```diff
@@ -993,8 +1039,15 @@ int main(int argc, char *argv[])
 	 * notleak() since our leak-detector can't find orig_argv on the
 	 * stack. */
 	orig_argv = notleak(tal_arr(ld, char *, argc + 1));
-	for (size_t i = 1; i < argc; i++)
+
+	for (size_t i = 1; i < argc; i++){
 		orig_argv[i] = tal_strdup(orig_argv, argv[i]);
+		if (strstarts(orig_argv[i], "--recover")) {
+			payload = tal(NULL, struct recover_payload);
+			payload->codex32secret = strchr(orig_argv[i], '=') + 1;
+		}
```

Comment:
Please add a `recover_payload` to `struct lightningd`, and set it inside the option parsing.  Or maybe just a `recover` string, and assemble the struct recover_payload (if required) in the branch below?

Reply:
You cannot do argument parsing here.  You need to do it inside opt_set_codex32.

That means you have to put a field inside `struct lightningd` so you know to call plugin_hook_call_recover.